### ISSUE TITLE
[Bug] devcontainer usage 

### ISSUE THREAD

**Issue by bartclarkson** [temporalio/samples-go#304]
### What are you really trying to do?

Goal: use the vscode devcontainer implementation with a minimum of configuration/overrides/additions.

Got things working as described below.  Looking for guidance on these "fixes", and suspecting the documentation should be updated to be more clear on these points.

### Describe the bug/repro/fixes

There seem to be two things required to get examples to run.  

1. The docker network ("temporal") is not configured for use as the DefaultHostPort by the Temporal Go SDK, which means the default `127.0.0.1:7233` produces failure until its correctly addressed.  The following modification describes the change necessary to every `c, err := client.Dial(client.Options{})` encountered in the examples. 

```
	clientOptions := client.Options{
		HostPort: "temporal:7233",
	}
	c, err := client.Dial(clientOptions)
```

Is there a way to globally set the `HostPort` default to `temporal:7233`?

2. To get the containers to all start, I added empty `development_es.yaml` at `<repo>/.devcontainer/dynamicconfig/development_es.yaml`. 

Thank you for this repository.

### Environment/Versions

OSX Monterey, Intel Chip

**Comment by cretz** [comment]
> Is there a way to globally set the HostPort default to temporal:7233?

No. We intentionally do not share code across samples and it probably doesn't have much value to change every sample to use something like an environment variable. The samples are meant as code samples for you to alter as needed to fit your use case or target host (same for the myriad of other client connection options you may have to change).

> To get the containers to all start, I added empty `development_es.yaml`

Hrmm, I wonder why this is needed. The docker compose should be enough. May have to investigate further to see why that is not the case.

**Comment by bartclarkson** [comment]
Fair enough, @cretz, and thank you for the response.

I didn't really intend to monkey around with devcontainers in VSCode.  The side quest proved entirely too interesting to resist, and has sparked some discussions on my end about the viability of embracing the practice.  It did in this case present the uninitiated engineer with some off-topic docker stuff to figure out, but such is life.
